### ISSUE TITLE
Code Help

### ISSUE THREAD

**Issue by TW5775** [ylwhxht/V2X-R#6]
Why does the dataset load as 0 when the code runs after changing root_ir and validate_dir according to what you wrote？

**Comment by TW5775** [comment]
![Image](https://github.com/user-attachments/assets/ab2dc517-d7e4-4a68-ac43-e347fd0ffda0)

**Comment by TW5775** [comment]
I am very interested in your project and want to run it

**Comment by ylwhxht** [comment]
> Why does the dataset load as 0 when the code runs after changing root_ir and validate_dir according to what you wrote？

Please provide some more information about the data in your root_dir and validate_dir paths, e.g. folder structure etc.

**Comment by ylwhxht** [comment]
> ![Image](https://github.com/user-attachments/assets/e46d6be1-4230-4bff-9f75-4e938894c1fa)
> 
> ![Image](https://github.com/user-attachments/assets/2370d017-9c93-48a2-8ec6-8a70d85ff8ca) Your reply was timely. Thank you very much

Need a little more detail on what exactly TRAIN and VALIDATE are in there. Otherwise, it's hard to pinpoint the problem😂

**Comment by TW5775** [comment]
![Image](https://github.com/user-attachments/assets/246722c4-a984-48fa-b71b-f03ac6ed7773)

![Image](https://github.com/user-attachments/assets/8f399aef-f522-4f43-b530-a131208f0772)

![Image](https://github.com/user-attachments/assets/cc7c9a94-f679-4e04-9e63-2596de3c5f2d)
/home/adlab5/ssd_3t/bigdata_3t/V2X-R/train
/home/adlab5/ssd_3t/bigdata_3t/V2X-R/validate

**Comment by ylwhxht** [comment]
> ![Image](https://github.com/user-attachments/assets/246722c4-a984-48fa-b71b-f03ac6ed7773)
> 
> ![Image](https://github.com/user-attachments/assets/8f399aef-f522-4f43-b530-a131208f0772)
> 
> ![Image](https://github.com/user-attachments/assets/cc7c9a94-f679-4e04-9e63-2596de3c5f2d) /home/adlab5/ssd_3t/bigdata_3t/V2X-R/train /home/adlab5/ssd_3t/bigdata_3t/V2X-R/validate

And inside, click on any date folder.

**Comment by TW5775** [comment]
![Image](https://github.com/user-attachments/assets/0d1b849b-ad2a-47a5-bc77-b634dc0e0bb5)

![Image](https://github.com/user-attachments/assets/4927babc-db37-4ded-8368-6242d4f392bd)

**Comment by ylwhxht** [comment]
> ![Image](https://github.com/user-attachments/assets/0d1b849b-ad2a-47a5-bc77-b634dc0e0bb5)
> 
> ![Image](https://github.com/user-attachments/assets/4927babc-db37-4ded-8368-6242d4f392bd)

OK, I'm back. It looks like there was a bug in the Decompression code that caused the agent number folder to be duplicated by one layer, which is 182/182/xxxxx.pcd in your picture.
We've just updated the Decompression code in the readme, please refresh and reuse the code to unzip it again

**Comment by ylwhxht** [comment]
> ![Image](https://github.com/user-attachments/assets/0d1b849b-ad2a-47a5-bc77-b634dc0e0bb5)
> 
> ![Image](https://github.com/user-attachments/assets/4927babc-db37-4ded-8368-6242d4f392bd)

@TW5775 Have you solved the problem yet?😊

**Comment by TW5775** [comment]
Sorry, I forgot to reply. The data set has been loaded successfully. Wish you all the best in your scientific research.😊

**Comment by ylwhxht** [comment]
> Sorry, I forgot to reply. The data set has been loaded successfully. Wish you all the best in your scientific research.😊

Well, you're welcome. You too, good luck with your research.

**Comment by TW5775** [comment]
Hello I'm here again, 😂the results of my test seem to be a little different from what I expected, may I ask if I have overlooked any details

![Image](https://github.com/user-attachments/assets/3e333f6b-4729-46f8-9cb9-88010d0875e8)

**Comment by ylwhxht** [comment]
> Hello I'm here again, 😂the results of my test seem to be a little different from what I expected, may I ask if I have overlooked any details
> 
> ![Image](https://github.com/user-attachments/assets/4a0517c3-c56f-4cfd-bb33-0e64d41fc5da)
> 
> ![Image](https://github.com/user-attachments/assets/ddac494b-3c0a-45cb-ba8c-12b26bca410d)
> 
> ![Image](https://github.com/user-attachments/assets/fdaa130c-abc5-41b4-bb55-b8f7fe25d484)
> 
> ![Image](https://github.com/user-attachments/assets/de7a1532-58fa-47bd-bf35-967671ed3d53)

What weather conditions did this experiment result in?

**Comment by ylwhxht** [comment]
> Hello I'm here again, 😂the results of my test seem to be a little different from what I expected, may I ask if I have overlooked any details
> 
> ![Image](https://github.com/user-attachments/assets/4a0517c3-c56f-4cfd-bb33-0e64d41fc5da)
> 
> ![Image](https://github.com/user-attachments/assets/ddac494b-3c0a-45cb-ba8c-12b26bca410d)
> 
> ![Image](https://github.com/user-attachments/assets/fdaa130c-abc5-41b4-bb55-b8f7fe25d484)
> 
> ![Image](https://github.com/user-attachments/assets/de7a1532-58fa-47bd-bf35-967671ed3d53)

I'm going to take a break, in the meantime, I've just re-trained based on the released codebase and waiting to see if the results are consistent.

Until then, the following comments are for your consideration:
1. The result of your reasoning is on the validation set, if the result in your graph is foggy/snowy then please switch to the test set and reason again, if it's for normal weather then even if it's on the validation set this result is equally abnormal.
2. We suggest you download the pre-trained model we provided first to see if you can reason out consistent results.
3. I just encountered the problem of loss reporting error while retraining is that this version of the code does not maintain the consistent number of cavs for de_lidar (due to the point cloud of some CAVs after FOV are empty, so the number of CAV may be inconsistent on different modalities). We just revised and updated it, but I'm curious as to how you were able to train on the previous version, did you fix the bug yourself (which could also have contributed to the problem)

**Comment by TW5775** [comment]
![Image](https://github.com/user-attachments/assets/97d2e9a3-1aec-47a4-ad35-361b8e63ea1e)

![Image](https://github.com/user-attachments/assets/be6742f0-58a6-4256-838d-0c038afe040a)

![Image](https://github.com/user-attachments/assets/9209bb27-05af-4274-b33c-bd38d3edbdc2)

![Image](https://github.com/user-attachments/assets/45c33f06-0648-4131-bdf7-f851ddde922b)Hello, I'm supposed to be training under the V2XR_AttFuse_with_MDD of foggy. I'm running these two pieces of code separately, and I see you've set up the configuration file. Do you think this needs to be changed?

**Comment by ylwhxht** [comment]
> ![Image](https://github.com/user-attachments/assets/97d2e9a3-1aec-47a4-ad35-361b8e63ea1e)
> 
> ![Image](https://github.com/user-attachments/assets/be6742f0-58a6-4256-838d-0c038afe040a)
> 
> ![Image](https://github.com/user-attachments/assets/9209bb27-05af-4274-b33c-bd38d3edbdc2)
> 
> ![Image](https://github.com/user-attachments/assets/45c33f06-0648-4131-bdf7-f851ddde922b)Hello, I'm supposed to be training under the V2XR_AttFuse_with_MDD of foggy. I'm running these two pieces of code separately, and I see you've set up the configuration file. Do you think this needs to be changed?

Did you use the newest code version？ And can you check the self.proj_first=True in intermediate_fusion_dataset？

If these is nothing wrong, try to first evaluate our pretrained Model follow our Readme to ensure the right of environment.

**Comment by TW5775** [comment]
I see, is this place missing parameters  proj_first ？

**Comment by TW5775** [comment]
Well, according to your method, I set this parameter to ture 
In this case, the result of eval_sim: false sim_weather: _fog_0.060 validate_dir: /home/adlab5/ssd_3t/bigdata_3t/v2x-r/validate in the configuration file is basically the same as in foggy. please ask the test situation in normal and snow conditions. Thank you😊
![Image](https://github.com/user-attachments/assets/bcee05df-233b-421b-ae07-8e20ed23f5b1)

**Comment by ylwhxht** [comment]
**Q1：the test situation in normal and snow conditions?**
A1：
normal: eval_sim = false (whatever the sim_weather) 
fog: eval_sim = true, sim_weather = _fog_0.060
snow: eval_sim = true, sim_weather = _snow_2.5_2.0

> Maybe you will feel puzzled, but just read continue reading.

---

**Q2：why eval_sim: false at this point is really the result of bad weather?**
A2:
The fact is that your result is not the result of foggy weather, it is the result of normal weather. But because you are using a validation set, the results are not quite the same as our normal weather results on test set (but they happen to be similar to the results in foggy weather). So if you need to compare the results with ours, please modify validate_dir from "xxxx/validate" to "xxxx/test".

---

**Q3：graphics card explodes**
A3:
Have you modified the batch size? I trained using the latest version of the code and the default configuration of Attfusie_with_SDD without any GPU error issues (we are using Nvidia RTX 3090 24GB).

Please check the batch size, the recommended size is 2. And ensure that the complete code is the latest version.

Also, please confirm if anyone else is running the program on the same GPU, as the changes in CAVs caused by collaborative perception result in variable memory usage.


---

**Additional explanation**
Previously, the model was trained and tested on different versions of code on different machines. In order to open source, we have integrated it into the current version. However, this has also led to fluctuations in the performance of some models (but<2%). We are re evaluating all models based on this code version and updating the corrected performance within a week. Therefore, if there are some fluctuations in the results, it is normal, but if the difference is significant, it is unreasonable.

**Comment by TW5775** [comment]
According to your method, I modified the test path and obtained the test results under normal, but when I set eval_sim = true and sim_weather = _fog_0.060, the test results were still the same as normal.😂
![Image](https://github.com/user-attachments/assets/86a2bcef-7395-4152-b26d-cd3d9bd2da37)
And the issue of graphics card explosions does not affect reproduction.😊

**Comment by ylwhxht** [comment]
> According to your method, I modified the test path and obtained the test results under normal, but when I set eval_sim = true and sim_weather = _fog_0.060, the test results were still the same as normal.😂 ![Image](https://github.com/user-attachments/assets/86a2bcef-7395-4152-b26d-cd3d9bd2da37) And the issue of graphics card explosions does not affect reproduction.😊

Strangely, it seems like I haven't updated the [base_dataset ](https://github.com/ylwhxht/V2X-R/blob/master/opencood/data_utils/datasets/opv2v/basedataset.py) that uses config to configure weather simulation, but I remember uploading it. Maybe there was a problem somewhere, and it was my mistake.
Please replace the latest version of the [base_dataset](https://github.com/ylwhxht/V2X-R/blob/master/opencood/data_utils/datasets/opv2v/basedataset.py) code and re-evaluate.
However, it should be noted that this may mean that your previous train_sim did not work, so simulated fog data may not have been used during training. You may need to re-train. The code related to calling simulated data is all in [base_dataset](https://github.com/ylwhxht/V2X-R/blob/master/opencood/data_utils/datasets/opv2v/basedataset.py), and it can be output in the corresponding position to confirm the use of simulated weather data (lines 190-193, 330-334)

**Comment by TW5775** [comment]
![Image](https://github.com/user-attachments/assets/7bdc0c5e-66fe-4cb1-9768-1417ae262518)
I retrained once after updating the code. However, eval_sim = true and sim_weather = _fog_0.060 are correct, but eval_sim = False, both result 0😂
